### PR TITLE
Bugs/addondocs

### PR DESCRIPTION
**Pull request by youngjeong46**
*Issue #, if available:* #334, #336, #337, #338, #339

*Description of changes:*
- Update all Addon Docs
- Listed the following in Index: Node Termination Handler, EFS CSI Driver, Fluent Bit, OPA Gatekeeper
- Labeled Karpenter and OPA as "Not Currently Supported, In Progress", as the recent update broke the addons (fix WIP)
- Make Usage code uniform across all addons
- change `ExternalDnsAddon ` to `ExternalDnsAddOn` to conform to uniform addon names

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/addons/app-mesh.md [hunk 1]
```diff
@@ -7,18 +7,23 @@ Full documentation on using App Mesh with EKS [can be found here](https://docs.a
...
-  env: {    
-      account: <AWS_ACCOUNT_ID>,
-      region: <AWS_REGION>,
-  },
-});
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
+
+const addOn = new blueprints.addons.AppMeshAddOn();
```
Comment (askulkarni2) [inline docs/addons/app-mesh.md:19]: Nitpick: `const appMeshAddOn` ?
Same for all the addons below as well.

File: docs/addons/app-mesh.md [hunk 2]
```diff
@@ -7,18 +7,23 @@ Full documentation on using App Mesh with EKS [can be found here](https://docs.a
...
+const addOns: Array<blueprints.ClusterAddOn> = [ addOn ];
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .account(account) 
+  .region(region)
+  .addOns(addOns)
+  .teams().build(app, 'my-stack-name', {env});
```
Comment (askulkarni2) [inline docs/addons/app-mesh.md:26]: Nit: `.build(..)` on newline for convention?
Same for all docs below.

File: lib/addons/external-dns/index.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ const defaultProps = {
  * Implementation of the External DNS service: https://github.com/kubernetes-sigs/external-dns/.
  * It is required to integrate with Route53 for external DNS resolution. 
  */
-export class ExternalDnsAddon extends HelmAddOn {
+export class ExternalDnsAddOn extends HelmAddOn {
```
Comment (askulkarni2) [inline lib/addons/external-dns/index.ts:33]: Nice catch!

**Review by askulkarni2 — DISMISSED** [review]
Minor comments, not blocking.

File: docs/addons/app-mesh.md (hunk 2 above)
Comment (shapirov103) [inline docs/addons/app-mesh.md:26]: minor: Drop the empty teams(). Applies elsewhere

Comment (shapirov103) [inline docs/addons/app-mesh.md:26]: {env} is not needed if you specify region and account on the blueprint. Applies elsewhere.

File: docs/addons/argo-cd.md [hunk 4]
```diff
@@ -13,18 +13,23 @@ Full Argo CD project documentation [can be found here](https://argoproj.github.i
...
 const app = new cdk.App();
-new EksBlueprint(app, 'my-stack-name', addOns, [], {
-  env: {    
-      account: <AWS_ACCOUNT_ID>,
-      region: <AWS_REGION>,
-  },
-});
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
```
Comment (shapirov103) [inline docs/addons/argo-cd.md:23]: let's drop this const altogether. You are setting account and region explicitly.

File: docs/addons/aws-for-fluent-bit.md [hunk 5]
```diff
@@ -7,18 +7,23 @@ AWS provides a Fluent Bit image with plugins for both CloudWatch Logs and Kinesi
...
 const app = new cdk.App();
-new EksBlueprint(app, 'my-stack-name', addOns, [], {
-  env: {
-      account: <AWS_ACCOUNT_ID>,
-      region: <AWS_REGION>,
-  },
-});
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
```
Comment (shapirov103) [inline docs/addons/aws-for-fluent-bit.md:17]: let's drop the env const.

File: docs/addons/aws-load-balancer-controller.md [hunk 6]
```diff
@@ -10,18 +10,23 @@ For more information about AWS Load Balancer Controller please see the [official
...
-      account: <AWS_ACCOUNT_ID>,
-      region: <AWS_REGION>
-  },
-});
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
+
+const addOn = new blueprints.addons.AwsLoadBalancerControllerAddOn();
+const addOns: Array<blueprints.ClusterAddOn> = [ addOn ];
```
Comment (shapirov103) [inline docs/addons/aws-load-balancer-controller.md:23]: this is not needed, let's drop this const. You can just pass addOn  to the builder:
```
.addOns(addOn)
```

File: docs/addons/cluster-autoscaler.md [hunk 7]
```diff
@@ -8,18 +8,23 @@ The Cluster Autoscaler add-on adds support for [Cluster Autoscaler](https://gith
...
-      account: <AWS_ACCOUNT_ID>,
-      region: <AWS_REGION>,
-  },
-});
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
+
+const addOn = new blueprints.addons.ClusterAutoscalerAddOn();
+const addOns: Array<blueprints.ClusterAddOn> = [ addOn ];
```
Comment (shapirov103) [inline docs/addons/cluster-autoscaler.md:21]: drop the const , env and empty teams

File: docs/addons/index.md [hunk 8]
```diff
@@ -12,20 +12,24 @@ The framework currently supports the following add-ons.
...
+| [`KarpenterAddOn (Currently Not Supported, In Progress)`](./karpenter.md)                                     | Adds [Karpenter](https://github.com/awslabs/karpenter) support for Amazon EKS.                                                        | 
 | [`KubeProxyAddOn`](./kube-proxy.md)                                    | Adds kube-proxy Amazon EKS add-on. Kube-proxy maintains network rules on each Amazon EC2 node                                         |
 | [`MetricsServerAddOn`](./metrics-server)                               | Adds metrics server (pre-req for HPA and other monitoring tools)                                                                      |
 | [`NginxAddOn`](./nginx.md)                                             | Adds NGINX ingress controller                                                                                                         |
+| [`OpaGatekeeperAddOn (Currently Not Supported, In Progress)`](./opa-gatekeeper.md)                            | Adds OPA Gatekeeper                                                                                                        |
```
Comment (shapirov103) [inline docs/addons/index.md:32]: 👍

**Review by shapirov103 — COMMENTED** [review]
Thank you for this PR. Some comments, mostly apply as a  pattern for builder-based blueprints.

**Review by shapirov103 — CHANGES_REQUESTED** [review]
Per comments

File: docs/addons/app-mesh.md (hunk 2 above)
Comment (youngjeong46) [reply docs/addons/app-mesh.md]: addressed.

File: docs/addons/argo-cd.md (hunk 4 above)
Comment (youngjeong46) [reply docs/addons/argo-cd.md]: addressed.

File: docs/addons/aws-for-fluent-bit.md (hunk 5 above)
Comment (youngjeong46) [reply docs/addons/aws-for-fluent-bit.md]: addressed.

File: docs/addons/aws-load-balancer-controller.md (hunk 6 above)
Comment (youngjeong46) [reply docs/addons/aws-load-balancer-controller.md]: addressed.

File: docs/addons/aws-node-termination-handler.md [hunk 9]
```diff
@@ -13,42 +13,29 @@ NTH can operate in two different modes: Instance Metadata Service (IMDS) or the
...
-}
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+const account = <AWS_ACCOUNT_ID>;
+const region = <AWS_REGION>;
+const env: { account, region },
+
+const addOn = new blueprints.addons.AwsNodeTerminationHandlerAddOn();
+const addOns: Array<blueprints.ClusterAddOn> = [ addOn ];
```
[the remark this answers is not in the document]
Comment (youngjeong46) [reply docs/addons/aws-node-termination-handler.md]: addressed.

File: docs/addons/secrets-store.md [hunk 10]
```diff
@@ -54,24 +55,12 @@ export class TeamBurnham extends ApplicationTeam {
...
-        // create this secret first
-        clusterInfo.cluster.node.addDependency(secret);
-        return secret
-    }
-}
-
-const app = new cdk.App();
-const teams: Array<ApplicationTeam> = [ new TeamBurnham(app) ];
-new EksBlueprint(app, 'my-stack-name', addOns, teams});
+const blueprint = blueprints.EksBlueprint.builder()
```
[the remark this answers is not in the document]
Comment (youngjeong46) [reply docs/addons/secrets-store.md:53]: calling it from the built-in. added "blueprints."

File: docs/addons/cluster-autoscaler.md [hunk 11]
```diff
@@ -57,7 +56,7 @@ The first step is to create a sample application via deployment and request 20m
 ```bash
 kubectl create deployment php-apache --image=us.gcr.io/k8s-artifacts-prod/hpa-example
 kubectl set resources deploy php-apache --requests=cpu=20m 
-kubectl expose php-apache --port 80
+kubectl expose deployment php-apache --port 80
```
Comment (shapirov103) [inline docs/addons/cluster-autoscaler.md:59]: nice!

File: docs/addons/nginx.md [hunk 12]
```diff
@@ -12,20 +12,20 @@ This add-on depends on [AWS Load Balancer Controller](aws-load-balancer-controll
...
-  },
-});
+
+const externalDnsHostname = ...;
+const awsLbControllerAddOn = new blueprints.addons.AwsLoadBalancerControllerAddOn();
+const nginxAddOn = new blueprints.addons.NginxAddOn({ externalDnsHostname })
+const addOns: Array<blueprints.ClusterAddOn> = [ awsLbControllerAddOn, nginxAddOn ];
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOns)
```
Comment (shapirov103) [inline docs/addons/nginx.md:27]: For multiple addons use spread operator (e.g. passing an array):
```
.addOns(...addOns)
```

**Pull request merged**